rel/server/overlays/bin/start_runtime.exs

```diff
@@ -0,0 +1,42 @@
+File.cd!(System.fetch_env!("HOME"))
+
+%{
+  node_base: node_base,
+  cookie: cookie,
+  dist_port: dist_port
+} = System.fetch_env!("LIVEBOOK_RUNTIME") |> Base.decode64!() |> :erlang.binary_to_term()
+
+# This is the only Fly-specific part of starting Livebook as runtime
+app = System.fetch_env!("FLY_APP_NAME")
+machine_id = System.fetch_env!("FLY_MACHINE_ID")
+node = :"#{node_base}@#{machine_id}.vm.#{app}.internal"
+
+# We persist the information before the node is reachable
+:persistent_term.put(:livebook_runtime_info, %{
+  pid: self(),
+  elixir_version: System.version()
+})
```

Comment:
If you want, you could move this to the runtime itself? And then you put the runtime module inside LIVEBOOK_RUNTIME env var and call it here? Then this whole code becomes generic?

Reply:
Btw, we should talk about the initialization flow. :)